protocol/verification.go

```diff
@@ -0,0 +1,56 @@
+package protocol
+
+import (
+	"bytes"
+
+	"github.com/coniks-sys/coniks-go/crypto/sign"
+	"github.com/coniks-sys/coniks-go/crypto/vrf"
+	m "github.com/coniks-sys/coniks-go/merkletree"
+)
+
+func verifyAuthPath(uname string, key []byte,
+	ap *m.AuthenticationPath,
+	str *m.SignedTreeRoot) ErrorCode {
+
+	// verify VRF Index
+	vrfKey := vrf.PublicKey(str.Policies.VrfPublicKey)
+	if !vrfKey.Verify([]byte(uname), ap.LookupIndex, ap.VrfProof) {
+		return ErrorBadVRFProof
+	}
+
+	if bytes.Equal(ap.LookupIndex, ap.Leaf.Index) { /* proof of inclusion */
+		// verify name-to-key binding
+		if !ap.VerifyBinding(key) {
+			return ErrorBadBinding
+		}
+		// verify commitment
+		if !ap.Leaf.Commitment.Verify([]byte(uname), ap.Leaf.Value) {
+			return ErrorBadCommitment
+		}
+	}
+
+	// verify auth path
+	if !ap.Verify(str.TreeHash) {
+		return ErrorBadMapping
+	}
+
+	return Passed
+}
+
+func verifySTR(signKey sign.PublicKey,
+	str *m.SignedTreeRoot,
+	savedSTR []byte) ErrorCode {
+
+	// verify STR's signature
+	if !signKey.Verify(str.Serialize(), str.Signature) {
+		return ErrorBadSignature
+	}
+
+	// verify hash chain
+	if str.Epoch > 0 && savedSTR != nil &&
```

Comment:
We already need to distribute the signing key through some other methods.  In the hangout, we said maybe we'd bake that into the client.  So, we could do the same thing.  Hardcode it in Tor Messenger.